segmentation_labeling_app/transforms/transformations.py

```diff
@@ -1,195 +1,114 @@
+            passed to numpy.random.default_rng if strategy is 'random'
 
     Returns:
-        new_video: A new video down-sampled to desired FPS with selected
-        strategy (time, row, col)
+        array_out: numpy.ndarray
+            array downsampled along axis=0
     """
     if output_fps > input_fps:
         raise ValueError('Output FPS cannot be greater than input FPS')
-    downsampled_ratio = input_fps / output_fps
-    downsampled_bin_cnt = len(full_array) / downsampled_ratio
+    if (strategy == 'maximum') & (len(array.shape) > 1):
+        raise ValueError("downsampling with strategy 'maximum' is not defined")
+
+    npts_in = array.shape[0]
+    npts_out = int(npts_in * output_fps / input_fps)
+    bin_list = np.array_split(np.arange(npts_in), npts_out)
+
+    array_out = np.zeros((npts_out, *array.shape[1:]))
 
-    downsampled_bins = np.array_split(full_array, downsampled_bin_cnt)
-    new_frames = np.zeros(shape=(len(downsampled_bins), full_array.shape[1],
-                                 full_array.shape[2]))
     if strategy == 'random':
-        random.seed(random_seed)
-
-    for i, downsampled_bin in enumerate(downsampled_bins):
-        if strategy == 'random':
-            rand_idx = random.randint(0, len(downsampled_bin) - 1)
-            new_frames[i] = downsampled_bin[rand_idx]
-        elif strategy == 'maximum':
-            new_frames[i] = np.max(downsampled_bin, axis=0)
-        elif strategy == 'average':
-            new_frames[i] = np.mean(downsampled_bin, axis=0)
-        elif strategy == 'first':
-            new_frames[i] = downsampled_bin[0]
-        elif strategy == 'last':
-            new_frames[i] = downsampled_bin[len(downsampled_bin) - 1]
-    return new_frames
-
-
-def get_transformed_center(coordinate_pair: Tuple[int, int],
-                           box_size: Tuple[int, int],
-                           video_shape: Tuple[int, int]) -> Tuple[int, int]:
-    """
-    Returns a coordinate pair that has been transformed to fit as closely to
-    original position as possible but still with box contained totally in video.
-    This is made to shift the coordinate if the subset box goes out of bound of
-    the video shape. The shift moves the box fully into the frame while maintaining
-    the center of the visual box is as close as possible to the original center.
-    (example coordinate pair at 0, with box size 3 x 3)
-    x x x x    x x x x
-    x x x x    x x x x
-    x x x x -> x x x x
-    x x x x    x 0 x x
-    x 0 x x    x x x x
-    Args:
-        coordinate_pair: the original coordinate location for centering (row, col)
-        box_size: the size of the box for subset (row, col)
-        video_shape: the x and y dimensions of the video (time, row, col)
+        rng = np.random.default_rng(random_seed)
 
-    Returns:
-        transformed_center: returns the best fitting center of the box for out
-        of bounds coordinates (row, col)
+        def sampler(arr, idx):
```

Comment:
Nit:
Redefining `sampler` is a little confusing. Since these can all be done in one line, might be clearer to use a dictionary?

```
sampling_strategies = {
    "maximum": lambda arr, idx: arr[idx].max(axis=0),
    "random": arr[np.random.default_rng(random_seed).choice(idx)],
...
}
sampler = sampling_strategies[strategy]
for i, bin_indices in enumerate(bin_list):
    array_out[i] = sampler(array, bin_indices)
```

Reply:
yeah, I agree, this doesn't look nice. Initially I was naming lambdas for shorter code, but flake8 did not like this.I was expecting some good suggestions here. I like the looks of the dictionary, I'll do that.

Reply:
Yeah, I see the purpose behind not naming lambdas but, I think it sometimes it could make things clearer. The dictionary is a good alternative (and also it's nice to select based on key).

Reply:
implemented